src/pages.c

```diff
@@ -833,9 +833,16 @@ init_thp_state(void) {
 	} else {
 		goto label_error;
 	}
+	if (opt_hpa_opts.hugify_style == hpa_hugify_style_auto) {
+		opt_hpa_opts.hugify_style = (init_system_thp_mode
+		                                == thp_mode_default)
+		    ? hpa_hugify_style_lazy
+		    : hpa_hugify_style_none;
+	}
 	return;
 #elif defined(JEMALLOC_HAVE_MEMCNTL)
 	init_system_thp_mode = thp_mode_default;
+	opt_hpa_opts.hugify_style = hpa_hugify_style_lazy;
```

Comment:
I think this mode is actually equivalent to eager since there is no equivalent to `MADV_NOHUGEPAGE` at least in the JEMalloc code.

https://docs.oracle.com/cd/E88353_01/html/E37841/memcntl-2.html